pycloudlib/gce/cloud.py

```diff
@@ -37,16 +43,26 @@ def __init__(
             timestamp_suffix: bool set True to append a timestamp suffix to the
                 tag
             credentials_path: path to credentials file for GCE
+                Defaults to default gcloud path.
             project: GCE project
+                Defaults to project supplied in 'credentials_path'
             region: GCE region
             zone: GCE zone
         """
         super().__init__(tag, timestamp_suffix)
         self._log.debug('logging into GCE')
 
-        if credentials_path:
+        credentials_path = Path(credentials_path)
+        if credentials_path.exists():
             os.environ["GOOGLE_APPLICATION_CREDENTIALS"] = str(
                 credentials_path)
+            if project is None:
+                with credentials_path.open() as f:
+                    config = json.load(f)
+                if 'project_id' in config:
```

Comment:
I'm not sure this is correct either; in both my desktop and in-container ADC file, I have only 4 keys: `client_id`, `client_secret`, `refresh_token`, and `type`.

When prompted to set my project in the container, it was set in `.config/gcloud/configurations/config_default` (which appears to be selected by `default` being in `.config/gcloud/active_config`).

(I believe that this file is used for authenticating your user to the Google APIs, which is orthogonal to the project being used: users have access to many projects.)